apps/meteor/app/api/server/v1/rooms.ts

```diff
@@ -333,6 +334,36 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'rooms.mediaEdit/:rid/:fileId',
+	{ authRequired: true, validateParams: isMediaEditProps },
+	{
+		async post() {
+			if (!(await canAccessRoomIdAsync(this.urlParams.rid, this.userId))) {
+				return API.v1.forbidden();
+			}
+
+			if (!this.bodyParams.fileName) {
+				throw new Meteor.Error('invalid-file-name');
+			}
+
+			const file = await Uploads.findOneById(this.urlParams.fileId);
+
+			if (!file) {
+				throw new Meteor.Error('invalid-file');
+			}
```

Comment:
```suggestion
```
If you're not gonna use the file data here, there's no need to load it. `updateFileNameById` won't match anything if the file doesn't exist and the error will be the same.

Alternatively when validating if a record exists you can use a projection to not load any of the record data.

Reply:
Done